internal/ctrlflow/transform.go

```diff
@@ -0,0 +1,212 @@
+package ctrlflow
+
+import (
+	"go/token"
+	"go/types"
+	mathrand "math/rand"
+	"strconv"
+
+	"golang.org/x/tools/go/ssa"
+)
+
+type blockMapping struct {
+	Fake, Target *ssa.BasicBlock
+}
+
+// applyFlattening adds a dispatcher block and uses ssa.Phi to redirect all ssa.Jump and ssa.If to the dispatcher,
+// additionally shuffle all blocks
+func applyFlattening(ssaFunc *ssa.Function, obfRand *mathrand.Rand) {
+	if len(ssaFunc.Blocks) < 3 {
+		return
+	}
+
+	phiInstr := &ssa.Phi{Comment: "ctrflow.phi"}
+	setType(phiInstr, types.Typ[types.Int])
+
+	entryBlock := &ssa.BasicBlock{
+		Comment: "ctrflow.entry",
+		Instrs:  []ssa.Instruction{phiInstr},
+	}
+	setBlockParent(entryBlock, ssaFunc)
+
+	makeJumpBlock := func(from *ssa.BasicBlock) *ssa.BasicBlock {
+		jumpBlock := &ssa.BasicBlock{
+			Comment: "ctrflow.jump",
+			Instrs:  []ssa.Instruction{&ssa.Jump{}},
+			Preds:   []*ssa.BasicBlock{from},
+			Succs:   []*ssa.BasicBlock{entryBlock},
+		}
+		setBlockParent(jumpBlock, ssaFunc)
+		return jumpBlock
+	}
+
+	// map for track fake block -> real block jump
+	var blocksMapping []blockMapping
+	for _, block := range ssaFunc.Blocks {
+		existInstr := block.Instrs[len(block.Instrs)-1]
+		switch existInstr.(type) {
+		case *ssa.Jump:
+			targetBlock := block.Succs[0]
+			fakeBlock := makeJumpBlock(block)
+			blocksMapping = append(blocksMapping, blockMapping{fakeBlock, targetBlock})
+			block.Succs[0] = fakeBlock
+		case *ssa.If:
+			tblock, fblock := block.Succs[0], block.Succs[1]
+			fakeTblock, fakeFblock := makeJumpBlock(tblock), makeJumpBlock(fblock)
+
+			blocksMapping = append(blocksMapping, blockMapping{fakeTblock, tblock})
+			blocksMapping = append(blocksMapping, blockMapping{fakeFblock, fblock})
+
+			block.Succs[0] = fakeTblock
+			block.Succs[1] = fakeFblock
+		case *ssa.Return, *ssa.Panic:
+			continue
```

Comment:
I think you don't need the `continue`?

Reply:
Why? panic or or return is ok, just flattening is not applicable to them and i ignore them

Reply:
I mean that without the continue line the behavior is exactly the same, unless I'm misreading the code